src/benchmark.md

```diff
@@ -0,0 +1,62 @@
+# Benchmark Report
+
+This benchmark report presents an analysis of the circuits and provable functions used in Anon Aadhaar. It includes constraint counts for provable functions (extractors and nullifier) and `ZkProgram` circuits. For provable functions, the `Provable.constraintSystem` API is used. Execution time for these functions is not measured, as they are currently executed outside of a circuit. They are planned to be used within the Aadhaar circuit, and for now, only their constraints are reported.
+
+## Device Information
+
+- CPU: Apple M2
+- Cores: 8
+- RAM: 16.00 GB
+- Node.js Version: 23.5.0
+- o1js Version: 2.4.0
+
+## Methodology
+
+Two different approaches are used for measurement:
+- `Provable.constraintSystem` is used to analyze the constraint count of functions like extractors or the nullifier. Only constraints are measured, as these functions are standalone and their execution time cannot be determined outside of a circuit. To obtain accurate constraint values, inputs to these functions are provided using `Provable.witness`.
+- `.analyzeMethods()` is used on `ZkProgram` circuits to extract constraint information in a structured format. To measure compile and execution times of `ZkProgram` methods, `performance.now()` is called before and after each computation.
+
+To minimize the impact of cache optimizations during circuit compilation, the `forceRecompile` option is enabled.
```

Comment:
Please elaborate more on the issue we faced and also why the current approach is taken